quantization/nlp/bert/migraphx/e2e_migraphx_bert_example.py

```diff
@@ -0,0 +1,484 @@
+    )
+
+    parser.add_argument(
+        "--version",
+        required=False,
+        default=1.1,
+        help='Squad dataset version. Default is 1.1. Choices are 1.1 and 2.0',
+        type=float
+    )
+
+    parser.add_argument(
+        "--no_eval",
+        action="store_true",
+        required=False,
+        default=False,
+        help='Turn off evaluate output result for f1 and exact match score. Default False',
+    )
+
+    parser.add_argument(
+        "--ort_verbose",
+        action="store_true",
+        required=False,
+        default=False,
+        help='Turn on onnxruntime verbose flags',
+    )
+
+    parser.add_argument(
+        "--ort_quant",
+        action="store_false",
+        required=False,
+        default=False,
+        help='Turn on Onnxruntime Quantizer instead of MIGraphX Quantizer',
+    )
+
+    parser.add_argument("--batch",
+                        required=False,
+                        default=1,
+                        help='Batch size per inference',
+                        type=int)
+
+    parser.add_argument("--seq_len",
+                        required=False,
+                        default=384,
+                        help='sequence length of the model. Default is 384',
+                        type=int)
+
+    parser.add_argument("--doc_stride",
+                        required=False,
+                        default=128,
+                        help='document stride of the model. Default is 128',
+                        type=int)
+
+    parser.add_argument("--cal_num",
+                        required=False,
+                        default=100,
+                        help='Number of calibration for QDQ Quantiation in int8. Default is 100',
+                        type=int)
+
+    parser.add_argument("--samples",
+                        required=False,
+                        default=0,
+                        help='Number of samples to test with. Default is 0 (All the samples in the dataset)',
+                        type=int)
+
+
+    parser.add_argument(
+        "--verbose",
+        action="store_true",
+        required=False,
+        default=False,
+        help='Show verbose output',
+    )
+    
+    return parser.parse_args()
+
+
+if __name__ == '__main__':
+    '''
+    BERT QDQ Quantization for MIGraphX.
+
+    There are two steps for the quantization,
+    first, calibration is done based on SQuAD dataset to get dynamic range of floating point tensors in the model
+    second, Q/DQ nodes with dynamic range (scale and zero-point) are inserted to the model
+
+    The onnx model used in the script is converted from MLPerf,
+    https://zenodo.org/records/3733910
+    
+    Some utility functions for dataset processing, data reader and evaluation are from Nvidia TensorRT demo BERT repo,
+    https://github.com/NVIDIA/TensorRT/tree/master/demo/BERT
+    
+    '''
+
+    flags = parse_input_args()
+
+    if flags.samples <= flags.batch:
```

Comment:
This will fail if `flags.samples` has the default `0` value, but it should not, because it means all the samples.
Maybe change this to:
```py
 if flags.samples and flags.samples <= flags.batch:
```
Also I think the errror message in the next line is wrong. According to the condition sample count can't be smaller than the batch size.